src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DefaultLSPDocumentMappingProvider.cs

```diff
@@ -95,25 +87,21 @@ public async override Task<Location[]> RemapLocationsAsync(Location[] locations,
                 }
 
                 var razorDocumentUri = RazorLSPConventions.GetRazorDocumentUri(uri);
-                if (!_documentManager.TryGetDocument(razorDocumentUri, out var documentSnapshot))
-                {
-                    continue;
-                }
 
                 var mappingResult = await MapToDocumentRangesAsync(
                     languageKind,
                     razorDocumentUri,
                     new[] { location.Range },
                     cancellationToken).ConfigureAwait(false);
 
-                cancellationToken.ThrowIfCancellationRequested();
-
-                if (mappingResult == null || mappingResult.HostDocumentVersion != documentSnapshot.Version)
```

Comment:
I would still keep this for documents the document manager does know about. I'd just change the logic to assume updated for any documents it doesn't know about

Reply:
Makes sense. Updated